File: test-omdb-api.py
import requests
import json
import urllib.parse

# Your OMDB API key
API_KEY = "d471685f"

# Test search query
SEARCH_TERM = "Guardians of the Galaxy"
ENCODED_SEARCH_TERM = urllib.parse.quote(SEARCH_TERM)

# Test with search
print(f"Testing OMDB API search for: {SEARCH_TERM}")
url = f"http://www.omdbapi.com/?apikey={API_KEY}&s={ENCODED_SEARCH_TERM}&type=movie"
print(f"URL: {url}")
response = requests.get(url)
search_data = response.json()
print(json.dumps(search_data, indent=2))

# Test with a specific movie ID
IMDB_ID = "tt3896198"  # Guardians of the Galaxy Vol. 2
print(f"\nTesting OMDB API for specific movie: {IMDB_ID}")
url = f"http://www.omdbapi.com/?apikey={API_KEY}&i={IMDB_ID}"
print(f"URL: {url}")
response = requests.get(url)
movie_data = response.json()
print(json.dumps(movie_data, indent=2))

# Test with a specific movie title
MOVIE_TITLE = "Guardians of the Galaxy Vol. 2"
ENCODED_MOVIE_TITLE = urllib.parse.quote(MOVIE_TITLE)
print(f"\nTesting OMDB API for specific movie title: {MOVIE_TITLE}")
url = f"http://www.omdbapi.com/?apikey={API_KEY}&t={ENCODED_MOVIE_TITLE}"
print(f"URL: {url}")
response = requests.get(url)
title_data = response.json()
print(json.dumps(title_data, indent=2)) 

File: TestOmdbApi.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;
import java.net.URLEncoder;

public class TestOmdbApi {
    public static void main(String[] args) {
        try {
            // Your OMDB API key
            String apiKey = "d471685f";
            
            // Test search query
            String searchTerm = "Guardians of the Galaxy";
            String encodedSearchTerm = URLEncoder.encode(searchTerm, "UTF-8");
            
            // Construct the URL
            String urlString = "http://www.omdbapi.com/?apikey=" + apiKey + "&s=" + encodedSearchTerm + "&type=movie";
            System.out.println("Testing URL: " + urlString);
            
            // Create URL and open connection
            URL url = new URL(urlString);
            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
            connection.setRequestMethod("GET");
            
            // Get response code
            int responseCode = connection.getResponseCode();
            System.out.println("Response Code: " + responseCode);
            
            if (responseCode == 200) {
                // Read the response
                BufferedReader reader = new BufferedReader(
                    new InputStreamReader(connection.getInputStream()));
                
                StringBuilder response = new StringBuilder();
                String line;
                while ((line = reader.readLine()) != null) {
                    response.append(line);
                }
                reader.close();
                
                // Print the response
                System.out.println("Response: " + response.toString());
            } else {
                System.out.println("Error: " + responseCode);
            }
            
            // Test with a specific movie ID
            String imdbId = "tt3896198"; // Guardians of the Galaxy Vol. 2
            String urlString2 = "http://www.omdbapi.com/?apikey=" + apiKey + "&i=" + imdbId;
            System.out.println("\nTesting URL for specific movie: " + urlString2);
            
            // Create URL and open connection
            URL url2 = new URL(urlString2);
            HttpURLConnection connection2 = (HttpURLConnection) url2.openConnection();
            connection2.setRequestMethod("GET");
            
            // Get response code
            int responseCode2 = connection2.getResponseCode();
            System.out.println("Response Code: " + responseCode2);
            
            if (responseCode2 == 200) {
                // Read the response
                BufferedReader reader = new BufferedReader(
                    new InputStreamReader(connection2.getInputStream()));
                
                StringBuilder response = new StringBuilder();
                String line;
                while ((line = reader.readLine()) != null) {
                    response.append(line);
                }
                reader.close();
                
                // Print the response
                System.out.println("Response: " + response.toString());
            } else {
                System.out.println("Error: " + responseCode2);
            }
            
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
} 